### PR TITLE
Break !role list in multiple messages if too long

### PR DESCRIPTION
**Pull request by wellme**
lmao ts

I have no idea if this works. Feel free to ignore this if you don't feel like testing.

### CODE REVIEW

File: src/messageHandlers/roleManager.ts [hunk 1]
```diff
@@ -139,19 +141,22 @@ class RoleManager implements MessageHandler {
...
-					let roleString = "Roles: "
-					const sortedRoles = msg.guild.roles.sort((x, y) => {
-						if (x.members.size > y.members.size)
-							return -1;
-						if (y.members.size > x.members.size)
-							return 1;
-						return 0;
-					});
+					let roleString = "Roles: ";
+					let queue = new Promise(res => res());
```
Comment (Yen) [inline src/messageHandlers/roleManager.ts]: What is the goal of this queue variable over awaiting the responder when required?

Comment (wellme) [reply src/messageHandlers/roleManager.ts]: Letting the rest of the messages build while one is being sent. There's a non-trivial chance that what I wrote is straight up wrong. My async is rusty.

Comment (Yen) [reply src/messageHandlers/roleManager.ts]: Yeah, this code does not make any sense and will actually do the reverse of what you want.
What you are doing is building up a queue of messages to post, then posting them all at the end. Where as what you just said you wanted was to send the messages while the others were being sent.
You can do this but that would require actually doing async stuff which is a pain, please just put `await responder(...` whenever you have built a new message to send.

Comment (wellme) [reply src/messageHandlers/roleManager.ts]: Tested with this sample https://gist.github.com/c6ad453c68140c0e0fb9552b6d4b880c (also tried without the sleep in the loop)
It seems to be doing what I want it to do. The items in the queue are sent as they are added (unless there's a backlog) because the promise is already fulfilled.

Comment (Yen) [reply src/messageHandlers/roleManager.ts]: No, there is some really dangerous stuff going on here, like the fact that you are capturing a local variable in these .then calls, only to then modify the local varible after before execution of the chained promise is guaranteed to have occurred. This is a race condition, and I hope that alone is enough to signify that this is really bad practice for how to write maintainable code.
On the other hand, this will give you no perfomance benefit, the asynchronous calls for these operations wont start as as the promise chain will not execute until flow control is yielded back to the caller. As you are only doing this at the very end with `await queue`, nothing happens asynchronously during the calculation of the remaining messages.
Basically, this is far too complex and does not actually do anything. Please convert this back to use a more understandable flow control by calling `await responder` when a message is ready to be sent so as to lower funciton complexity.

Comment (wellme) [reply src/messageHandlers/roleManager.ts]: Yeah, ok. I see why this was wrong. Thanks for the explanation. I still don't see why this is a race condition (The value of the string objects aren't changed, they are immutable), but whatever.

Comment (Yen) [reply src/messageHandlers/roleManager.ts]: It is a race condition because you are not capturing the value of the string, you are capturing the local variable which is a reference. When you call `.then`, it queues up a function to run after the previous promise has resolved. This includes the often overlooked operation of actually getting the value a named varible references. As such, by changing the `queue` variable after queuing up the promise to chain, it will cause the callback passed into `.then` to get whatever value `queue` has at the time it actually gets evaluated, _not_ at the time you called `.then`. It's a pretty easy thing to miss but a good reason why we try and avoid messing around with manual management of asynchronous control flow. :)
